helpers/withInstance.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect } from 'react'
+
+const withInstance = client => {
+  const [instance, setInstance] = useState()
+  const [context, setContext] = useState()
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const instancePromise = client
+          .getStackClient()
+          .fetchJSON('GET', '/settings/instance')
+
+        const contextPromise = client
+          .getStackClient()
+          .fetchJSON('GET', '/settings/context')
+        const promises = [instancePromise, contextPromise]
+
+        const [instanceFetched, contextFetched] = await Promise.all(
+          promises.map(p => p.catch(e => e))
+        )
+        if (!(instanceFetched instanceof Error)) {
+          setInstance(instanceFetched)
+        }
+        if (!(contextFetched instanceof Error)) {
+          setContext(contextFetched)
+        }
+      } catch (e) {} //eslint-disable-line
+    }
+    fetchData()
+  }, [])
```

Comment:
I think we should have a long cache for context and instance, inside cozy-client.

Reply:
Yep, I think we should have a long cache for those. https://github.com/cozy/cozy-client/issues/551